streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/impl/KubernetesNativeSessionSubmit.scala

```diff
@@ -152,8 +152,9 @@ object KubernetesNativeSessionSubmit extends KubernetesNativeSubmitTrait with Lo
       val kubernetesClusterDescriptor = getK8sClusterDescriptorAndSpecification(flinkConfig)
       clusterDescriptor = kubernetesClusterDescriptor._1
       kubeClient = FlinkKubeClientFactory.getInstance.fromConfiguration(flinkConfig, "client")
-
-      if (deployRequest.clusterId != null && kubeClient.getRestService(deployRequest.clusterId).isPresent) {
+      // after flink 1.15 FlinkKubeClient removed the getRestService method
+      if (deployRequest.clusterId != null && hasMethod(kubeClient.getClass, "getRestService")
```

Comment:
This code logic is strange,

e.g: `if ( a.hasMethod("bbb") && a.bbb() )`
if a has no bbb method, Next, call the bbb method,  The code compiles will fail.

Reply:
The compilation used is 1.14, the actual run will load the corresponding version of the dependency, after 1.15 removed the method getRestService, so it will fail. I will look at the detailed logic of flink again and implement it under optimization

Reply:
Thanks @monrg 's contribution.

It's similar to `FlinkClientTrait`, we can add the `FlinkKubeClientTrait` in the `streampark-flink-shims-base` module, it depends flink-1.14 by default.

Every flink-shims extends the `FlinkKubeClientTrait` and call the `FlinkKubeClient#getRestService before flink 1.15` or the `FlinkKubeClient#getService since flink-1.15`.

And KubernetesNativeSessionSubmit always calls the `FlinkKubeClientTrait#getService`, don't care call `getService` or `getRestService`.